README.md

```diff
@@ -28,62 +28,103 @@ Usage of cluster-proportional-autoscaler:
 The code in this module is a Kubernetes Golang API client that, using the default service account credentials
 available to Golang clients running inside pods, it connects to the API server and polls for the number of nodes
 and cores in the cluster.
+
 The scaling parameters and data points are provided via a ConfigMap to the autoscaler and it refreshes its
 parameters table every poll interval to be up to date with the latest desired scaling parameters.
 
 ## Calculation of number of replicas
 
-The desired number of replicas is computed by looking up the number of cores and nodes using the step ladder function.
-The step ladder functions use the datapoint for core and node scaling from the configmap.
-The lookup which yields the higher number of replicas will be used as the target scaling number.
-This may be later extended to more complex interpolation or linear/exponential scaling schemes
-but it currently supports (and defaults to) to this step mode only.
+The desired number of replicas is computed by using the number of cores and nodes as input of the chosen controller.
+
+This may be later extended to more complex interpolation or exponential scaling schemes
+but it currently supports `linear` and `ladder` modes(default to `linear`).
 
-# Configmap controlling parameters
+# Control patterns and ConfigMap formats
 
 The ConfigMap provides the configuration parameters, allowing on-the-fly changes without rebuilding or
 restarting the scaler containers/pods.
 
-Currently the only supported ConfigMap key value is: ladder, which is also the only supported controll mode.
+Currently the two supported ConfigMap key value is: `ladder` and `linear`, which corresponding to two supported control mode.
+
+## Linear Mode
+
+Parameters in ConfigMap must be JSON and use `linear` as key. The sub-keys as below indicates:
+
+```
+data:
+  linear: |-
+    { 
+      "coresPerReplica": 2,
+      "nodesPerReplica": 1,
+      "min": 1,
+      "max": 100
+    }
+```
+
+The idea of linear control mode is simple: how many cores or nodes could be taking care by one replica?
```

Comment:
I think you should put the equation and then give the example

replicas = max( cores * 1/coresPerReplica , nodes * 1/nodesPerReplica )